src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFactoryAdapter.java

```diff
@@ -46,6 +46,7 @@
 /**
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
+@Deprecated
```

Comment:
why deprecate it if you are still using it in a new method?

Reply:
Because we don't want to have a public class creating KubernetesClients from outside the (protected) Provider. In the future, we will at least reduce the visibility of the factoryAdapter.

Reply:
In this case, you should add `@Restricted(NoExternalUse.class)` to prevent any actual usage outside the plugin, and also add a short comment to motivate the deprecation to potential users.